dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/user/UserController.java

```diff
@@ -816,13 +823,31 @@ protected void prePatchEntity( User oldEntity, User newEntity )
     }
 
     @Override
-    protected void postPatchEntity( User user )
+    protected void postPatchEntity( JsonPatch patch, User entityAfter )
     {
         // Make sure we always expire all the user's active sessions if we
         // have disabled the user.
-        if ( user != null && user.isDisabled() )
+        if ( entityAfter != null && entityAfter.isDisabled() )
         {
-            userService.expireActiveSessions( user );
+            userService.expireActiveSessions( entityAfter );
+        }
+
+        if ( entityAfter != null && patch != null )
+        {
+            for ( JsonPatchOperation op : patch.getOperations() )
+            {
+                JsonPointer userGroups = op.getPath().matchProperty( "userGroups" );
+                String opName = op.getOp();
+                if ( userGroups != null && opName.equals( "add" ) )
```

Comment:
Check for `replace` also, as in this case they have the same semantics

Reply:
ok